APMrover2/APMrover2.cpp

```diff
@@ -398,44 +392,23 @@ void Rover::update_GPS_10Hz(void)
 {
     have_position = ahrs.get_position(current_loc);
 
-    if (gps.last_message_time_ms() != last_gps_msg_ms && gps.status() >= AP_GPS::GPS_OK_FIX_3D) {
+    if (gps.last_message_time_ms() != last_gps_msg_ms) {
         last_gps_msg_ms = gps.last_message_time_ms();
 
-        if (ground_start_count > 1) {
-            ground_start_count--;
-
-        } else if (ground_start_count == 1) {
-            // We countdown N number of good GPS fixes
-            // so that the altitude is more accurate
-            // -------------------------------------
-            if (current_loc.lat == 0 && current_loc.lng == 0) {
-                ground_start_count = 20;
-            } else {
-                init_home();
+        // set system time if necessary
+        set_system_time_from_GPS();
```

Comment:
yes indeed.  Copter does it this way though so it's consistent.  Perhaps there's some small advantage of having the set-system-time-from-GPS call independent of the other stuff.  In fact, the other stuff shouldn't even depend upon the GPS, it should depend upon the AHRS/EKF but that's an existing issue that we can fix another time.